services/head_tracker.go

```diff
@@ -175,73 +181,114 @@ func (ht *HeadTracker) onNewHead(head *models.BlockHeader) {
 	}
 }
 
-func (ht *HeadTracker) subscribeToNewHeads(headers chan models.BlockHeader) (models.EthSubscription, error) {
-	sub, err := ht.store.TxManager.SubscribeToNewHeads(headers)
-	if err != nil {
-		return nil, err
+func (ht *HeadTracker) listenForNewHeads() {
+	defer ht.listenForNewHeadsWg.Done()
+	defer ht.unsubscribeFromHead()
+
+	for {
+		ht.resubscriptionPoll()
+
+	SubscriptionLoop:
+		for {
```

Comment:
What about something like `err = ht.receiveHeadersLoop(); if err != nil { continue } else { return }` rather than break to label?